File: androidApp/src/main/kotlin/org/kmptemplate/project/ui/components/AppSuccessCard.kt
package org.kmptemplate.project.ui.components

import androidx.compose.foundation.background
import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Box
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.Spacer
import androidx.compose.foundation.layout.fillMaxSize
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.height
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.layout.safeContentPadding
import androidx.compose.foundation.layout.size
import androidx.compose.foundation.shape.CircleShape
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material3.MaterialTheme
import androidx.compose.material3.OutlinedButton
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.text.style.TextAlign
import androidx.compose.ui.unit.sp
import org.kmptemplate.project.ui.theme.AppShapes
import org.kmptemplate.project.ui.theme.AppSizes
import org.kmptemplate.project.ui.theme.AppSpacing

/**
 * Kartu sukses dengan ikon bulat + title/subtitle/caption + pasangan tombol primer & sekunder.
 * Dipakai di layar hasil (post-login, post-register, walkthrough completed).
 */
@Composable
fun AppSuccessCard(
    title: String,
    subtitle: String,
    caption: String,
    primaryText: String,
    onPrimary: () -> Unit,
    secondaryText: String,
    onSecondary: () -> Unit,
    modifier: Modifier = Modifier,
    emoji: String = "🎉",
) {
    Column(
        modifier = modifier
            .fillMaxSize()
            .safeContentPadding()
            .padding(AppSpacing.xxl),
        horizontalAlignment = Alignment.CenterHorizontally,
        verticalArrangement = Arrangement.Center
    ) {
        Box(
            modifier = Modifier
                .size(AppSizes.successIcon)
                .clip(CircleShape)
                .background(MaterialTheme.colorScheme.primaryContainer),
            contentAlignment = Alignment.Center
        ) {
            Text(text = emoji, fontSize = 42.sp)
        }
        Spacer(Modifier.height(AppSpacing.xl))
        Text(
            title,
            style = MaterialTheme.typography.headlineMedium.copy(fontWeight = FontWeight.Bold),
            textAlign = TextAlign.Center
        )
        Spacer(Modifier.height(AppSpacing.sm + AppSpacing.xs / 2))
        Text(
            subtitle,
            style = MaterialTheme.typography.titleMedium,
            color = MaterialTheme.colorScheme.primary,
            textAlign = TextAlign.Center
        )
        Spacer(Modifier.height(AppSpacing.xs))
        Text(
            caption,
            style = MaterialTheme.typography.bodyMedium,
            color = MaterialTheme.colorScheme.onSurfaceVariant,
            textAlign = TextAlign.Center
        )
        Spacer(Modifier.height(AppSpacing.xxl + AppSpacing.xs))
        AppPrimaryButton(text = primaryText, onClick = onPrimary)
        Spacer(Modifier.height(AppSpacing.md))
        OutlinedButton(
            onClick = onSecondary,
            modifier = Modifier
                .fillMaxWidth()
                .height(AppSizes.buttonHeight),
            shape = RoundedCornerShape(AppShapes.cornerLarge),
        ) {
            Text(secondaryText)
        }
    }
}

File: iosApp/iosApp/UI/Components/AppSuccessCard.swift
import SwiftUI

/// Kartu sukses — ikon check bulat + title/subtitle/caption + pasangan tombol primer + sekunder.
/// Digunakan di layar hasil (post-login, post-register, walkthrough completed).
struct AppSuccessCard: View {
    let systemImage: String
    let title: String
    let subtitle: String
    let caption: String
    let primaryTitle: String
    let onPrimary: () -> Void
    let secondaryTitle: String
    let secondaryTint: Color
    let onSecondary: () -> Void

    init(
        systemImage: String = "checkmark.circle.fill",
        title: String,
        subtitle: String,
        caption: String,
        primaryTitle: String,
        onPrimary: @escaping () -> Void,
        secondaryTitle: String,
        secondaryTint: Color = .red,
        onSecondary: @escaping () -> Void
    ) {
        self.systemImage = systemImage
        self.title = title
        self.subtitle = subtitle
        self.caption = caption
        self.primaryTitle = primaryTitle
        self.onPrimary = onPrimary
        self.secondaryTitle = secondaryTitle
        self.secondaryTint = secondaryTint
        self.onSecondary = onSecondary
    }

    var body: some View {
        VStack(spacing: AppSpacing.xl) {
            ZStack {
                Circle()
                    .fill(Color.green.opacity(0.15))
                    .frame(width: AppSizes.successIcon, height: AppSizes.successIcon)

                Image(systemName: systemImage)
                    .font(.system(size: 52, weight: .medium))
                    .foregroundColor(.green)
            }
            .padding(.top, AppSpacing.sm)

            VStack(spacing: AppSpacing.sm) {
                Text(title)
                    .font(.title2.weight(.bold))
                    .foregroundColor(.primary)

                Text(subtitle)
                    .font(.headline)
                    .foregroundColor(.accentColor)

                Text(caption)
                    .font(.subheadline)
                    .foregroundColor(.secondary)
            }

            VStack(spacing: AppSpacing.md) {
                AppPrimaryButton(title: primaryTitle, systemImageAfter: nil, action: onPrimary)
                AppSecondaryButton(title: secondaryTitle, tint: secondaryTint, action: onSecondary)
            }
            .padding(.top, AppSpacing.md)
        }
        .padding(AppSpacing.xl + AppSpacing.xs)
        .background(Color(uiColor: .systemBackground))
        .clipShape(RoundedRectangle(cornerRadius: AppCornerRadius.xLarge, style: .continuous))
        .shadow(color: Color.black.opacity(0.06), radius: 16, x: 0, y: 6)
        .padding(.horizontal, AppSpacing.xl)
    }
}
